Source/WebKit/UIProcess/API/ios/WKWebViewIOS.mm

```diff
@@ -919,7 +919,7 @@ - (void)_updateScrollViewForTransaction:(const WebKit::RemoteLayerTreeTransactio
     }
 
     bool scrollingEnabled = _page->scrollingCoordinatorProxy()->hasScrollableOrZoomedMainFrame() || hasDockedInputView || isZoomed || scrollingNeededToRevealUI;
-    [_scrollView _setScrollEnabledInternal:scrollingEnabled];
+    [_scrollView panGestureRecognizer].allowedTouchTypes = scrollingEnabled ? @[@(UITouchTypeDirect), @(UITouchTypeIndirectPointer)] : @[ ];
```

Comment:
Can we use a default list of touch types in the `scrollingEnabled` case rather than specifying our own? The current logic could result in unexpected behavior if the default list changed.

Reply:
is there an easy way of getting the default list?

Reply:
Looks like there is an internal method (`_defaultAllowedTouchTypes`) which would give us the default list.

Given that the method isn't exposed, this might be the best approach for now:
1. Verify that this list matches the existing default.
2. Make an SPI request and add a FIXME here.

Alternatively, we could just stage the SPI and call the method, if we can confirm that UIKit is willing to expose it.

Reply:
according to <[https://developer.apple.com/documentation/uikit/uigesturerecognizer/1624223-allowedtouchtypes]()> this is the default list

i'll create a radar asking UIKit to expose `_defaultAllowedTouchTypes` as SPI :)

Reply:
> The default value of this property contains all touch types.

I think we will also need `UITouchTypeIndirect` and `UITouchTypePencil`?

Reply:
i coulda sworn reading somewhere that `UITouchTypeIndirectPointer` was a special enum case representing `UITouchTypeIndirect | UITouchTypePencil`, but now i can't find it 🤔